src/Compilers/CSharp/Portable/CSharpCompilationOptions.cs

```diff
@@ -494,6 +498,16 @@ public new CSharpCompilationOptions WithStrongNameProvider(StrongNameProvider pr
             return new CSharpCompilationOptions(this) { StrongNameProvider = provider };
         }
 
+        public new CSharpCompilationOptions WithSyntaxTreeOptionsProvider(SyntaxTreeOptionsProvider provider)
+        {
+            if (ReferenceEquals(provider, this.SyntaxTreeOptionsProvider))
```

Comment:
Hmm, maybe we use a different pattern outside the core.  Prefer style of the file.